lib/sensors/accelerometer.cpp

```diff
@@ -13,6 +42,44 @@ Accelerometer::~Accelerometer()
 {
 }
 
+std::optional<core::Result> Accelerometer::isValueReady()
+{
+  // check to see if the values are ready to be read
+  const auto data_ready = i2c_->readByte(kDefaultAccelerometerAddress, kDataReady);
+  if (!data_ready) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read acceleration data");
+    return std::nullopt;
+  }
+  if (*data_ready % 2 == 0) {
+    logger_.log(core::LogLevel::kWarn, "Failed to read acceleration data as it is not ready");
+    return core::Result::kFailure;
+  }
+  return core::Result::kSuccess;
+}
+
+// TODOLater: current settings of the accelerometer make it read in +-16g but with high noise. Check
+// whether other configuration could be better
+std::optional<core::RawAccelerationData> Accelerometer::read()
+{
+  const auto result_x = getRawAcceleration(core::Axis::kX);
+  if (!result_x) { return std::nullopt; }
+  const std::int32_t x_acceleration = getAccelerationFromRawValue(*result_x);
+  const auto result_y               = getRawAcceleration(core::Axis::kY);
+  if (!result_y) { return std::nullopt; }
+  const std::int32_t y_acceleration = getAccelerationFromRawValue(*result_y);
+  const auto result_z               = getRawAcceleration(core::Axis::kZ);
+  if (!result_z) { return std::nullopt; }
+  const std::int32_t z_acceleration = getAccelerationFromRawValue(*result_z);
+  const std::optional<core::RawAccelerationData> acceleration_3_axis{
+    std::in_place,
+    x_acceleration,
+    y_acceleration,
+    z_acceleration,
+    std::chrono::system_clock::now()};
```

Comment:
Would you not want to pass in a time source and use time_.now() to make the code testable?

Reply:
good catch! will change